pennylane/numpy/__init__.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This package provides a wrapped version of autograd.numpy, such that
+it works with the PennyLane :class:`~.tensor` class.
+"""
+# pylint: disable=wrong-import-position,wildcard-import,undefined-variable
+from autograd import numpy as _np
+from autograd.numpy import *
+
+from .wrapper import wrap_arrays, extract_tensors
+
+wrap_arrays(_np.__dict__, globals())
+
+# Delete the unwrapped fft, linalg, random modules

```

Comment:
What is special about these three submodules that they require special care?

Reply:
These are the 3 subpackages of NumPy that are supported by Autograd, so we need to re-wrap them to make sure that they support the new `tensor` subclass. 

I don't 100% like the solution here (deleting `fft`, `linalg`, and `random`, and then importing the wrapped versions), but I think there is subtlety to Python's import system I wasn't truly grasping. No matter how I ordered the imports, or where I did the wrapping, `pennylane.numpy.random` would always simply point to `autograd.numpy.random` from the `from autograd.numpy import *` line.

It's only when I delete those modules from `global()` before importing from the relative path does python properly import the wrapped modules.